main.js

```diff
@@ -1,251 +1,154 @@
+// instantiate nuxt.js with the options
+const nuxtConfig = require(`./nuxt.config.js`);
+const nuxt = new Nuxt(nuxtConfig);
 
-    if (modelId === undefined || modelId === ``) {
-      response.redirect(301, `/${manufacturer.key}`);
-      return;
-    }
+// render every remaining route with Nuxt.js
+app.use(nuxt.render);
 
-    if (modelId in register.rdm[manufacturerId].models) {
-      const locationHash = (personalityIndex === undefined || personalityIndex === ``) ? `` : `#rdm-personality-${personalityIndex}`;
-      response.redirect(301, `/${manufacturer.key}/${manufacturer.models[modelId]}${locationHash}`);
-      return;
-    }
-  }
-
-  response.status(404).render(`pages/rdm-not-found`, getOptions(request, {
-    manufacturerId: manufacturerId,
-    modelId: modelId
-  }));
-});
-
-app.get(`/sitemap.xml`, (request, response) => {
-  response.type(`application/xml`).render(`pages/sitemap`, getOptions(request));
-});
-
-// support json encoded bodies
-app.use(bodyParser.json());
-
-app.post(`/ajax/add-fixtures`, (request, response) => {
-  require(`./lib/add-fixtures.js`)(request, response);
-});
-
-// if no other route applies
-app.use((request, response, next) => {
-  // first char is always a slash, last char has to be tested
-  const sanitizedUrl = request.originalUrl.slice(1, (request.originalUrl.slice(-1) === `/`) ? -1 : request.originalUrl.length);
-  const segments = sanitizedUrl.split(`/`);
-
-  if (segments.length === 1) {
-    // manufacturer page
-    if (segments[0] in manufacturers) {
-      response.render(`pages/single_manufacturer`, getOptions(request, {
-        man: segments[0]
-      }));
-      return;
-    }
-
-    // download all
-    const [download, pluginName] = segments[0].split(`.`);
-    if (download === `download` && pluginName in plugins.export) {
-      const fixtures = Object.keys(register.filesystem).map(fixture => {
-        const [man, key] = fixture.split(`/`);
-        return fixtureFromRepository(man, key);
-      });
-      const outfiles = plugins.export[pluginName].export(fixtures, {
-        baseDir: __dirname
-      });
-
-      downloadFiles(response, outfiles, pluginName);
-      return;
-    }
-  }
-
-  if (segments.length === 2 && segments[0] in manufacturers) {
-    const man = segments[0];
-    const fix = segments[1];
-
-    // fixture page
-    if (register.manufacturers[man].includes(fix)) {
-      response.render(`pages/single_fixture`, getOptions(request, {
-        man: man,
-        fix: fix
-      }));
-      return;
-    }
-
-    // fixture export
-    const [key, pluginName] = fix.split(`.`);
-    if (register.manufacturers[man].includes(key) && pluginName in plugins.export) {
-      const outfiles = plugins.export[pluginName].export([fixtureFromRepository(man, key)], {
-        baseDir: __dirname
-      });
-
-      downloadFiles(response, outfiles, pluginName);
-      return;
-    }
-  }
+if (process.argv[2] === `--dev`) {
```

Comment:
Idea: What about using minimist and adding another argument to set the port? It could be useful to use another port only for this session (e.g. to run the same server twice). Our http status test could also easily use another port then, which allows you to run the test while the server is still running.

Reply:
Seems like a good idea, but I'll save this for another PR. See #439.